### PR TITLE
Remove the mixedType config for JSON as it has no downsides any longer

### PR DESCRIPTION
**Pull request by revans2**
mixedTypesAsStrings was added in when we relied on CUDF to do schema discovery when parsing JSON. That is no longer the case and there are no real downsides to it any longer. So remove the config because it is not really needed.

This patch sets it to always be on, as that is the direction we want to go in.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: docs/compatibility.md [hunk 1]
```diff
@@ -368,10 +368,8 @@ In versions of Spark before 3.5.0 there is no maximum to how deeply nested JSON
 no matter what version of Spark is used. If the nesting level is over this the JSON is considered
 invalid and all values will be returned as nulls.
 
-Only structs are supported for nested types. There are also some issues with arrays of structs. If
-your data includes this, even if you are not reading it, you might get an exception. You can
-try to set `spark.rapids.sql.json.read.mixedTypesAsString.enabled` to true to work around this,
-but it also has some issues with it.
+Mixed types can have some problem. If an item being read could have some lines that are arrays 
+and others that are structs/dictionaries it is possible for an error will be thrown.
```
Comment (jlowe) [inline docs/compatibility.md:372]:
```suggestion
Mixed types can have some problems. If an item being read could have some lines that are arrays 
and others that are structs/dictionaries it is possible an error will be thrown.
```

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala [hunk 2]
```diff
@@ -268,6 +268,7 @@ object GpuJsonReadCommon {
 
   private def throwMismatchException(cv: ColumnView,
       dt: DataType): (Option[ColumnView], Seq[AutoCloseable]) = {
+    ai.rapids.cudf.TableDebug.get.debug(s"JSON MISMATCH $dt", cv)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:271]: Intentional or debug code left in?

Comment (revans2) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonReadCommon.scala:271]: oops

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**